`openblas.pc` file in build `openblas-v0.3.20-571-g3dec11c6`  is incorrect on `win_amd64`

The scipy windows cibuildwheel entries in https://github.com/scipy/scipy/pull/18012 are failing with the OpenBLAS build that's available on anaconda. In the build log is the error message:

    c:/rtools40/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lopenblas
    collect2.exe: error: ld returned 1 exit status
which is unusual because meson said it found openblas. However, I then had a look at openblas.pc from the openblas-v0.3.20-571-g3dec11c6-win_amd64-gcc_10_3_0.zip:

```
libdir=/c/opt/64/lib
libsuffix=
includedir=/c/opt/64/include
openblas_config= USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS= NO_LAPACK= NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP= SKYLAKEX MAX_THREADS=2
version=0.3.21.dev
extralib=-defaultlib:advapi32 -lgfortran -defaultlib:advapi32 -lgfortran
Name: openblas
Description: OpenBLAS is an optimized BLAS library based on GotoBLAS2 1.13 BSD version
Version: ${version}
URL: https://github.com/xianyi/OpenBLAS
Libs: -L${libdir} -lopenblas${libsuffix}
Libs.private: ${extralib}
Cflags: -I${includedir}
```

I think meson sets up all the link commands using the Libs entry. Since the `libsuffix` is empty I think the library it's looking for in the link step is -lopenblas. However, that library isn't present, only `libopenblas_v0.3.20-571-g3dec11c6-gcc_10_3_0` is available. i.e. libsuffix needs to be fixed.

In comparison openblas.pc for v0.3.21 is `Libs: -L${libdir} -lopenblas_v0.3.21-gcc_10_3_0`, and that library IS present in lib.

--- a/tools/build_openblas.sh
+++ b/tools/build_openblas.sh
@@ -123,7 +123,11 @@ dlltool --input-def ${DLL_BASENAME}.def \
     --dllname ${DLL_BASENAME}.dll \
     --output-lib ${DLL_BASENAME}.lib
 # Replace the DLL name with the generated name.
-sed -i "s/ -lopenblas$/ -l${DLL_BASENAME:3}/g" pkgconfig/openblas.pc
+sed -i "s/ -lopenblas.*$/ -l${DLL_BASENAME:3}/g" pkgconfig/openblas*.pc
+echo After sed -i, pkgconfig/openblas*.pc is
+echo ------------
+cat pkgconfig/openblas*.pc
+echo ------------
 cd ../..
 # Build template site.cfg for using this build
 cat > ${build_bits}/site.cfg.template << EOF

